pancancer-arch-3/src/main/java/info/pancancer/arch3/jobGenerator/JobGenerator.java

```diff
@@ -225,7 +225,8 @@ private void enqueueNewJobs(String job) {
 
             this.jchannel.basicPublish("", queueName + "_orders", MessageProperties.PERSISTENT_TEXT_PLAIN,
                     job.getBytes(StandardCharsets.UTF_8));
-        } catch (IOException ex) {
+            jchannel.waitForConfirms();
+        } catch (IOException | InterruptedException ex) {
```

Comment:
Would it make sense to have handle the `InterruptedException` with a something more graceful here (such as a proper shutdown-and-cleanup method), or not?